api/src/nv_ingest_api/internal/transform/embed_text.py

```diff
@@ -78,23 +79,37 @@ def _make_async_request(
         # Normalize API key to avoid sending an empty bearer token via SDK internals
         _token = (api_key or "").strip()
         _api_key = _token if _token else "<no key provided>"
-        client = OpenAI(
-            api_key=_api_key,
-            base_url=embedding_nim_endpoint,
-        )
-
-        extra_body = {
-            "input_type": input_type,
-            "truncate": truncate,
-        }
-        if modalities:
-            extra_body["modality"] = modalities
-
-        resp = client.embeddings.create(
-            input=prompts,
-            model=embedding_model,
-            encoding_format=encoding_format,
-            extra_body=extra_body,
+        # client = OpenAI(
+        #     api_key=_api_key,
+        #     base_url=embedding_nim_endpoint,
+        # )
+
+        # extra_body = {
+        #     "input_type": input_type,
+        #     "truncate": truncate,
+        # }
+        # if modalities:
+        #     extra_body["modality"] = modalities
+
+        # resp = client.embeddings.create(
+        #     input=prompts,
+        #     model=embedding_model,
+        #     encoding_format=encoding_format,
+        #     extra_body=extra_body,
+        # )
+
+        resp = infer_microservice(
```

Comment:
For consistency, this really needs to be a NimClient, so that we're able to use the same configuration process that we do for all the other NIM elements.

Reply:
Something like this but, I would imagine much less pre/post processing code.

https://github.com/NVIDIA/nv-ingest/blob/main/api/src/nv_ingest_api/internal/primitives/nim/model_interface/yolox.py#L92